plugin/src/main/java/org/opensearch/ml/action/models/UpdateModelTransportAction.java

```diff
@@ -354,40 +426,84 @@ private void updateRequestConstructor(
-            }, e -> {
-                log
-                    .error(
-                        "Failed to register ML model with model ID {} to the new model group with model group ID {}",
-                        modelId,
-                        newModelGroupId
-                    );
-                actionListener.onFailure(e);
-            }));
+            if (isInPlaceUpdate) {
+                String[] targetNodeIds = getAllNodes();
+                MLInPlaceUpdateModelNodesRequest mlInPlaceUpdateModelNodesRequest = new MLInPlaceUpdateModelNodesRequest(
+                    targetNodeIds,
+                    modelId,
+                    isPredictorUpdate
+                );
+                client.update(updateModelGroupRequest, ActionListener.wrap(r -> {
+                    client
+                        .update(
+                            updateRequest,
+                            getUpdateResponseListenerWithInPlaceUpdate(modelId, wrappedListener, mlInPlaceUpdateModelNodesRequest)
+                        );
+                }, e -> {
+                    log
+                        .error(
+                            "Failed to register ML model with model ID {} to the new model group with model group ID {}",
+                            modelId,
+                            newModelGroupId
+                        );
+                    wrappedListener.onFailure(e);
+                }));
+            } else {
+                client.update(updateModelGroupRequest, ActionListener.wrap(r -> {
+                    client.update(updateRequest, getUpdateResponseListener(modelId, wrappedListener));
+                }, e -> {
+                    log
+                        .error(
+                            "Failed to register ML model with model ID {} to the new model group with model group ID {}",
+                            modelId,
+                            newModelGroupId
+                        );
+                    wrappedListener.onFailure(e);
+                }));
+            }
         } catch (IOException e) {
             log.error("Failed to build update request.");
-            actionListener.onFailure(e);
+            wrappedListener.onFailure(e);
         }
     }
 
-    private ActionListener<UpdateResponse> getUpdateResponseListener(
+    private ActionListener<UpdateResponse> getUpdateResponseListenerWithInPlaceUpdate(
         String modelId,
-        ActionListener<UpdateResponse> actionListener,
-        ThreadContext.StoredContext context
+        ActionListener<UpdateResponse> wrappedListener,
+        MLInPlaceUpdateModelNodesRequest mlInPlaceUpdateModelNodesRequest
     ) {
-        return ActionListener.runBefore(ActionListener.wrap(updateResponse -> {
+        return ActionListener.wrap(updateResponse -> {
+            if (updateResponse != null && updateResponse.getResult() != DocWriteResponse.Result.UPDATED) {
```

Comment:
Can we add a comment here for this check?

Reply:
Added